python/cugraph/structure/graph.py

```diff
@@ -969,24 +947,138 @@ def neighbors(self, n):
+                return cudf.Series(dtype="int")
             n = node[0]
 
         df = self.edgelist.edgelist_df
-        neighbors = df[df['src'] == n]['dst'].reset_index(drop=True)
+        neighbors = df[df["src"] == n]["dst"].reset_index(drop=True)
         if self.renumbered:
-            return self.edgelist.renumber_map.iloc[neighbors]
+            # FIXME:  Multi-column vertices
+            return self.renumber_map.from_vertex_id(neighbors)["0"]
         else:
             return neighbors
 
+    def unrenumber(self, df, column_name, preserve_order=False):
+        """
+        Given a DataFrame containing internal vertex ids in the identified
+        column, replace this with external vertex ids.  If the renumbering
+        is from a single column, the output dataframe will use the same
+        name for the external vertex identifiers.  If the renumbering is from
+        a multi-column input, the output columns will be labeled 0 through
+        n-1 with a suffix of _column_name.
+
+        Note that this function does not guarantee order in single GPU mode,
+        and does not guarantee order or partitioning in multi-GPU mode.  If you
+        wish to preserve ordering, add an index column to df and sort the
+        return by that index column.
+
+        Parameters
+        ----------
+        df: cudf.DataFrame or dask_cudf.DataFrame
+            A DataFrame containing internal vertex identifiers that will be
+            converted into external vertex identifiers.
+
+        column_name: string
+            Name of the column containing the internal vertex id.
+
+        preserve_order: (optional) bool
+            If True, preserve the order of the rows in the output
+            DataFrame to match the input DataFrame
+
+        Returns
+        ---------
+        df : cudf.DataFrame or dask_cudf.DataFrame
+            The original DataFrame columns exist unmodified.  The external
+            vertex identifiers are added to the DataFrame, the internal
+            vertex identifier column is removed from the dataframe.
+
+        Examples
+        --------
+        >>> M = cudf.read_csv('datasets/karate.csv', delimiter=' ',
+        >>>                   dtype=['int32', 'int32', 'float32'], header=None)
+        >>>
+        >>> df, number_map = NumberMap.renumber(df, '0', '1')
+        >>>
+        >>> G = cugraph.Graph()
+        >>> G.from_cudf_edgelist(df, 'src', 'dst')
+        >>>
+        >>> pr = cugraph.pagerank(G, alpha = 0.85, max_iter = 500,
+        >>>                       tol = 1.0e-05)
+        >>>
+        >>> pr = number_map.unrenumber(pr, 'vertex')
+        >>>
+        """
+        return self.renumber_map.unrenumber(df, column_name, preserve_order)
+
+    def lookup_vertex_id(self, df, column_name=None):
+        """
+        Given a DataFrame containing external vertex ids in the identified
+        columns, or a Series containing external vertex ids, return a
+        Series with the vertex ids.
```

Comment:
with the "internal" vertex ids?

I think "vertex_id" in look_vertex_id, add_vertex_id, to_vertex_id, from_vertex_id ... can be confusing as both external & internal vertex_IDs are both vertex IDs. Can I get little more explanation about this naming or can we bring better names?

Reply:
Each of these refer to internal vertex ids.  I could rename these functions to:
* lookup_internal_vertex_id
* add_internal_vertex_id
* to_internal_vertex_id
* from_internal_vertex_id
If that would introduce more clarity.

Unfortunately, one of my complaints about our python formatting standards, longer function and variable names make for more silly indenting due to the 80 character constraint on line length.

Reply:
If the goal is to make the name shorter, we may try something like "lookup_int_vid", but I guess this will raise consistency issue with the rest of cuGraph. If I need to pick between confusing names vs ugly formatting, I will go for ugly formatting.

Reply:
I'd rather get rid of the 80 character line limit.  Thought we got rid of punched cards at the beginning of my career :-)

I also prefer the longer names.  I was mostly whining.

Reply:
The character line limit is helpful for reviews on github and browsing code in split-window mode.
Now 80 characters look really limited to me. I have no objection increasing it after we double-check what was the initial motivation for this number. Pretty sure 120 would still be ok for github reviews. In comparison, C++ limit is 200.
Also agree that naming > formating